dask_planner/src/expression.rs

```diff
@@ -49,130 +58,69 @@ impl From<ScalarValue> for PyScalarValue {
+    /// function to take and create the PyExpr.
+    pub fn from(expr: Expr, input: Option<Arc<LogicalPlan>>) -> PyExpr {
+        PyExpr {
+            input_plan: input,
+            expr: expr,
+        }
     }
 
-    pub fn column_name(&self, mut plan: logical::PyLogicalPlan) -> String {
+    fn _column_name(&self, mut plan: LogicalPlan) -> String {
         match &self.expr {
             Expr::Alias(expr, name) => {
                 println!("Alias encountered with name: {:?}", name);
+                // let reference: Expr = *expr.as_ref();
+                // let plan: logical::PyLogicalPlan = reference.input().clone().into();
 
                 // Only certain LogicalPlan variants are valid in this nested Alias scenario so we
                 // extract the valid ones and error on the invalid ones
                 match expr.as_ref() {
                     Expr::Column(col) => {
                         // First we must iterate the current node before getting its input
-                        match plan.current_node() {
-                            LogicalPlan::Projection(proj) => match proj.input.as_ref() {
-                                LogicalPlan::Aggregate(agg) => {
-                                    let mut exprs = agg.group_expr.clone();
-                                    exprs.extend_from_slice(&agg.aggr_expr);
-                                    match &exprs[plan.get_index(col)] {
-                                        Expr::AggregateFunction { args, .. } => match &args[0] {
-                                            Expr::Column(col) => {
-                                                println!("AGGREGATE COLUMN IS {}", col.name);
-                                                col.name.clone()
+                        match plan {
+                            LogicalPlan::Projection(proj) => {
+                                match proj.input.as_ref() {
+                                    LogicalPlan::Aggregate(agg) => {
+                                        let mut exprs = agg.group_expr.clone();
+                                        exprs.extend_from_slice(&agg.aggr_expr);
+                                        let col_index: usize =
+                                            proj.input.schema().index_of_column(col).unwrap();
+                                        // match &exprs[plan.get_index(col)] {
+                                        match &exprs[col_index] {
+                                            Expr::AggregateFunction { args, .. } => {
+                                                match &args[0] {
+                                                    Expr::Column(col) => {
+                                                        println!(
+                                                            "AGGREGATE COLUMN IS {}",
+                                                            col.name
+                                                        );
```

Comment:
If the debug prints are useful as work progresses, it might be worth adding a `TODO` comment to clean up later

Reply:
Yep, my plan is to not merge with upstream master during all of these commits. What that will give us is basically another shot to review before we merge into main as all of these type changes will stick out like red flags and can be changed then but at the same time can be left in for now to assist with current development work.

Long story short. 100% plan on removing them during that final PR